core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -672,11 +672,24 @@ public BlockRegion getBlockRegion() {
       }
 
       public void finish() throws IOException {
-        try {
-          in.close();
-        } finally {
-          compressAlgo.returnDecompressor(decompressor);
-          decompressor = null;
+        final InputStream inLocal = in;
+        if (inLocal != null) {
+          synchronized (inLocal) {
+            if (in != null) {
+              try {
+                in.close();
+              } finally {
+                in = null;
```

Comment:
Could also use a boolean to track closed here and avoid the inLocal and double check.